src/shapes/lovestruck.js

```diff
@@ -0,0 +1,72 @@
+module.exports = function drawLovestruck(ctx) {
+  ctx.save();
+  ctx.fillStyle = "rgba(0, 0, 0, 0)";
+  ctx.beginPath();
+  ctx.moveTo(0,0);
+  ctx.lineTo(17,0);
+  ctx.lineTo(17,16);
+  ctx.lineTo(0,16);
+  ctx.closePath();
+  ctx.clip();
+  ctx.translate(0,0);
+  ctx.translate(0,0);
+  ctx.scale(1,1);
+  ctx.translate(0,0);
```

Comment:
You can clear out the default clipping and translation steps here if you want to optimize a bit!

Also the save/restore steps aren't always necessary, if the `fillStyle` is being specifically set to a new value. Just make sure they're balanced (equal number of save and restore calls).

Reply:
Which ones are the default clipping and translations?